test-packages/s4test/src/test/R/test.s4.dispatch.metadata.01.R

```diff
@@ -0,0 +1,97 @@
+#
+# Renjin : JVM-based interpreter for the R language for the statistical analysis
+# Copyright © 2010-2016 BeDataDriven Groep B.V. and contributors
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, a copy is available at
+# https://www.gnu.org/licenses/gpl-2.0.txt
+#
+
+library(hamcrest)
+library(methods)
+
+
+setClass("AA", representation(a="numeric"))
+setClass("BB", contains="AA")
+
+setMethod("Arith", signature("AA","AA"), function(e1, e2) environment())
+setMethod("^", signature("AA","AA"), function(e1, e2) environment())
+setMethod("*", signature("BB","BB"), function(e1, e2) environment())
+a <- new("AA")
+b <- new("BB")
+
+m = a + a
+n = a ^ a
+o = b + b
+p = b ^ b
+q = b * b
+
+
+assertThat(
+    ls(all.names=TRUE) %in% c(".Random.seed", "a", "b", ".__C__AA", ".__C__BB", "m", "n", "o",
+    "p", "q", ".__T__Arith:base", ".__T__^:base", ".__T__*:base"),
+    identicalTo( c(TRUE, TRUE, TRUE, TRUE, TRUE, TRUE, TRUE, TRUE, TRUE, TRUE, TRUE, TRUE, TRUE)  )
+    )
+
+test.s4.dispatch.metadata.01b = function() {
+    assertThat( ls(m, all.names=TRUE) %in% c(".defined", "e1", "e2", ".Generic", ".Method", ".Methods", ".target") , identicalTo( c(TRUE, TRUE, TRUE, TRUE, TRUE, TRUE, TRUE) ))
+}
+
+test.s4.dispatch.metadata.01c = function() {
+    assertThat( ls(n, all.names=TRUE) , identicalTo( c("e1", "e2") ))
+}
+
+test.s4.dispatch.metadata.01d = function() {
+    assertThat( ls(o, all.names=TRUE) %in% c(".defined", "e1", "e2", ".Generic", ".Method", ".Methods", ".target"), identicalTo( c(TRUE, TRUE, TRUE, TRUE, TRUE, TRUE, TRUE) ))
+}
+
+test.s4.dispatch.metadata.01e = function() {
+    assertThat( ls(p, all.names=TRUE) %in% c(".defined", "e1", "e2", ".Generic", ".Method", ".Methods", ".target") , identicalTo( c(TRUE, TRUE, TRUE, TRUE, TRUE, TRUE, TRUE) ))
+}
+
+test.s4.dispatch.metadata.01f = function() {
+    assertThat( ls(q, all.names=TRUE) %in% c("e1", "e2") , identicalTo( c(TRUE, TRUE) ))
+}
+
+test.s4.dispatch.metadata.02 = function() {
+    # typeof
+    assertThat( typeof(m$e1) , identicalTo( "S4"  ))
```

Comment:
Ideally you should use the output from GNU R here. 

In GNU R, evaluate:

```.R
cat(deparse(m$e1))
```
and then take that output and add the assertion:

```.R
assertThat(m$e1, identicalTo(structure(c(1L, 2L), .Names = c(...), RandoAttrib = ...)))
```